heroic-core/src/main/java/com/spotify/heroic/metric/LocalMetricManager.java

```diff
@@ -413,19 +433,13 @@ private void accept(final Consumer<MetricBackend> op) {
         final boolean failOnLimits;
 
         @Override
-        public void resolved(final Pair<Series, FetchData> result) throws Exception {
-            final FetchData f = result.getRight();
-            requestErrors.addAll(f.getErrors());
-            try {
-                for (final MetricCollection g : f.getGroups()) {
-                    g.updateAggregation(session, result.getLeft().getTags(),
-                        ImmutableSet.of(result.getLeft()));
-                    dataInMemoryReporter.reportDataNoLongerNeeded(g.size());
-                }
-            } catch (QuotaViolationException e) {
-                // Seems resolved() should not throw exception
-                log.debug("Aggregation quota violated", e);
-            }
+        public void resolved(final FetchData.Result result) throws Exception {
+            requestErrors.addAll(result.getErrors());
+        }
+
+        void addMetricsCollection(final Series series, MetricCollection g) {
```

Comment:
This is more of a consumer than a collection, so `acceptMetricCollection(..)` feels like a more appropriate name.